src/structures/APIMessage.js

```diff
@@ -166,6 +176,8 @@ class APIMessage {
     if (this.isMessage) {
       // eslint-disable-next-line eqeqeq
       flags = this.options.flags != null ? new MessageFlags(this.options.flags).bitfield : this.target.flags.bitfield;
+    } else if (this.isInteraction) {
+      flags = this.options.ephemeral ? MessageFlags.EPHEMERAL : undefined;
```

Comment:
```suggestion
      flags = this.options.ephemeral ? MessageFlags.FLAGS.EPHEMERAL : undefined;
      type = this.options.hideSource ? InteractionResponseType.CHANNEL_MESSAGE : InteractionResponseType.CHANNEL_MESSAGE_WITH_SOURCE;
```
missing a step & also allow to hide the source